crud/src/main/java/com/redhat/lightblue/eval/SetExpressionEvaluator.java

```diff
@@ -139,7 +139,9 @@ public SetExpressionEvaluator(JsonNodeFactory factory, FieldTreeNode context, Se
             } else if (mdNode instanceof ArrayField) {
                 data = initializeArray(rvalue, refMdNode, mdNode, field, refPath);
             }
-            setValues.add(data);
+            if (data != null) {
```

Comment:
this would work, but we should also add an if statement *before* instanceof ArrayField (becayse ResolvedReferenceField extends ArrayField)

if(mdNode instanceof ResolvedReferenceField) {
    // then, if the set-to value is null, set data to null, so it won't be added to setValues. If it is non-null, throw error, because the call is setting a reference to a non-null value.
}